solutions/process_orders.rb

```diff
@@ -0,0 +1,64 @@
+# Your boss comes out of their office and says:
+# "We need to chagne our Warehouse app. We now allow people to pay more money, to get their deliveries earlier."
+#
+# You need to take and process orders. Process just means tell the warehouse folk which order todo first from teh Warehouse.
+#
+# Orders come in as a Ruby Hash via Warehouse.receive and the lower the number, the high priority it gets.
+# So, if you have 2 orders... one being a priority 1, another being a priority 2... process the 1 first from the orders.
+
+# Example order:
+# order = { priority: 1,
+#           product: "boots",
+#           address: "633 Folsom St."
+# }
+
+class Warehouse
+  @@orders = {}
+
+  def self.orders
+    @@orders
+  end
+
+  def self.print_orders
+    p "Current orders: #{@@orders}"
+  end
+
+  # Add to orders
+  def self.recieve(order)
+    if @@orders[order[:priority]] == nil
```

Comment:
could these checks about the state of the orders/priority be abstracted into helpers? I can read the code easily, but it's difficult to imagine the state that you are describing.